Compfiles/Imo1959P4.lean

```diff
@@ -0,0 +1,21 @@
+/-
+Copyright (c) 2024 The Compfiles Contributors. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: InternLM-MATH LEAN Formalizer v0.1. This is auto-formalized by InternLM-MATH LEAN Formalizer v0.1, modified and verified by InternLM-MATH team members.
+-/
+
+import Mathlib.Tactic
+
+import ProblemExtraction
+
+problem_file { tags := [.Algebra] }
+
+/-!
+# International Mathematical Olympiad 1959, Problem 4
+
+Construct a right triangle with a given hypotenuse $c$ such that the median drawn to the hypotenuse is the geometric mean of the two legs of the triangle.
```

Comment:
My understanding is that "construct" in this context means [straightedge and compass construction](https://en.wikipedia.org/wiki/Straightedge_and_compass_construction). So framing this as a purely algebraic problem misses a big part of what's being asked for.